packages/zowe-explorer/src/trees/shared/SharedInit.ts

```diff
@@ -338,6 +338,37 @@ export class SharedInit {
                 ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.UPDATE);
             });
         });
+
+        try {
+            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(imperative.ZoweUserEvents.ON_VAULT_CHANGED, async () => {
+                ZoweLogger.info(vscode.l10n.t("Changes in the credential vault detected, refreshing Zowe Explorer."));
+                await ProfilesUtils.readConfigFromDisk();
+                await SharedActions.refreshAll(providers.ds);
+                await SharedActions.refreshAll(providers.uss);
+                await SharedActions.refreshAll(providers.job);
```

Comment:
After merging in changes from the next branch, should we replace this with a single call to `await SharedActions.refreshAll();` to refresh all providers?